pkg/keyspace/tso_keyspace_group.go

```diff
@@ -1013,6 +1013,85 @@ func (m *GroupManager) FinishMergeKeyspaceByID(mergeTargetID uint32) error {
 	return nil
 }
 
+// MergeAllIntoDefaultKeyspaceGroup merges all other keyspace groups into the default keyspace group.
+func (m *GroupManager) MergeAllIntoDefaultKeyspaceGroup() error {
+	defer logutil.LogPanic()
+	mergedGroupNum := 0
+	for i := 0; i < int(endpoint.UserKindCount); i++ {
+		userKind := endpoint.UserKind(i)
+		log.Info("start to merge all keyspace groups into the default one",
+			zap.Stringer("user-kind", userKind))
+		groups, ok := m.groups[userKind]
+		if !ok || groups.Len() == 0 {
+			continue
+		}
+		var (
+			maxBatchSize  = MaxEtcdTxnOps/2 - 1
+			groupsToMerge = make([]uint32, 0, maxBatchSize)
+		)
+		for idx, group := range groups.GetAll() {
+			if group.ID == utils.DefaultKeyspaceGroupID {
+				continue
+			}
+			groupsToMerge = append(groupsToMerge, group.ID)
+			if len(groupsToMerge) < maxBatchSize && idx < groups.Len()-1 {
+				continue
+			}
+			log.Info("merge keyspace groups into the default one",
+				zap.Int("index", idx),
+				zap.Int("batch-size", len(groupsToMerge)),
+				zap.Int("merged-group-num", mergedGroupNum))
+			// Reach the batch size, merge them into the default keyspace group.
+			if err := m.MergeKeyspaceGroups(utils.DefaultKeyspaceGroupID, groupsToMerge); err != nil {
```

Comment:
Is there possible that some of the groups are merged and the rest are not?

Reply:
Yes, it's possible if the merging batch failed halfway. And it's better to let the API caller determine whether we should retry or do something else. IMO, maybe we could introduce the retry mechanism in `pd-ctl` later to ensure the merging could be finished as much as possible.

Reply:
How about printing the merged/unmerged count in this situation?

Reply:
Added, PTAL.